### DISCUSSION TITLE
Need to match a malformed URL generated by a client app

### DISCUSSION THREAD

**Discussion by bmatatuk** [dotnet/yarp#1997] [Q&A] [answered]
Hi 
I have a Reverse Proxy configured and is working fine. But, I found there is a bunch of client application that is generating a malformed URL. I found the URL has a double forward slash. 

The correct URL is https://customer1.domain.com/api/v1/connection/status. But one of the client applications is generating https://customer1.domain.com//api/v1/connection/status. If you notice, there are 2 forward slashes before **api**!

Is there any way I can match both of these variants (single or double slashes) in the path using regular expression? I do not want to duplicate the route.

My current code in the configuration is : 
`...
"Match": {
    "Path": "/api/{**catch-all}",
    "Hosts": [ "customer1.domain.com" ]
   },
...
`
I tried a few things with Regular Expression but didn't work. Any help is appreciated

Thanks

**Comment by MihaZupan** [comment, marked as answer]
> Is there any way I can match both of these variants (single or double slashes) in the path

The configuration you are using (`"/api/{**catch-all}"`) will already match both `/api/foo` and `/api//bar`.

YARP will forward these requests as-is. That is, the backend destination will also see `/api/foo` and `/api//bar`.
Are you looking to configure YARP such that it accepts both, but forwards them on in a normalized format (`/api/foo` and `/api/bar`)?

**Reply by bmatatuk** [reply]
Thanks for the response. 

If you see my post, the double slash is soon after the host and NOT further down the path. 

It match https://customer1.domain.com/api/v1/connection/status. But, it DOES NOT https://customer1.domain.com//api/v1/connection/status

**NOTE:** I have configured InterceptorFactory on the proxy app to test different scenarios.

Hope that make sense.

Thanks

**Reply by MihaZupan** [reply]
I don't know if there's a convenient way to achieve this in routing, but you can insert a small middleware before routing happens that corrects such requests.

For example, this will also handle `domain.com////api/foo`
```c#
app.Use((context, next) =>
{
    var path = context.Request.Path.Value.AsSpan();
    if (path.StartsWith("//", StringComparison.Ordinal))
    {
        context.Request.Path = new PathString(string.Concat("/", path.TrimStart('/')));
    }

    return next();
});

app.UseRouting();
```

**Reply by bmatatuk** [reply]
Fantastic solution... That worked like a charm!!!! Saved my day

Thank you so much for your help.
